code/game/machinery/computer/records/security.dm

```diff
@@ -0,0 +1,525 @@
+#define COMP_SECURITY_ARREST_AMOUNT_TO_FLAG 10
+#define PRINTOUT_MISSING "Missing"
+#define PRINTOUT_RAPSHEET "Rapsheet"
+#define PRINTOUT_WANTED "Wanted"
+/// Editing this will cause UI issues.
+#define MAX_CRIME_NAME_LEN 24
+
+/obj/machinery/computer/records/security
+	name = "security records console"
+	desc = "Used to view and edit personnel's security records."
+	icon_screen = "security"
+	icon_keyboard = "security_key"
+	req_one_access = list(ACCESS_SEC_RECORDS)
+	circuit = /obj/item/circuitboard/computer/records/security
+	light_color = LIGHT_COLOR_RED
+
+	/// The current state of the printer
+	var/printing = FALSE
+
+CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/computer/records/security)
+
+/obj/machinery/computer/records/security/syndie
+	icon_keyboard = "syndie_key"
+	req_one_access = list(ACCESS_SYNDICATE)
+
+/obj/machinery/computer/records/security/laptop
+	name = "security laptop"
+	desc = "A cheap Nanotrasen security laptop, it functions as a security records console. It's bolted to the table."
+	icon_state = "laptop"
+	icon_screen = "seclaptop"
+	icon_keyboard = "laptop_key"
+
+/obj/machinery/computer/records/security/laptop/syndie
+	desc = "A cheap, jailbroken security laptop. It functions as a security records console. It's bolted to the table."
+	req_one_access = list(ACCESS_SYNDICATE)
+
+/obj/machinery/computer/records/security/Initialize(mapload, obj/item/circuitboard/C)
+	. = ..()
+	AddComponent(/datum/component/usb_port, list(
+		/obj/item/circuit_component/arrest_console_data,
+		/obj/item/circuit_component/arrest_console_arrest,
+	))
+
+/obj/machinery/computer/records/security/emp_act(severity)
+	. = ..()
+
+	if(machine_stat & (BROKEN|NOPOWER) || . & EMP_PROTECT_SELF)
+		return
+
+	for(var/datum/record/crew/target in GLOB.manifest.general)
+		if(prob(10/severity))
+			switch(rand(1,5))
+				if(1)
+					if(prob(10))
+						target.name = "[pick(random_lizard_name(MALE),random_lizard_name(FEMALE))]"
+					else
+						target.name = "[pick(pick(GLOB.first_names_male), pick(GLOB.first_names_female))] [pick(GLOB.last_names)]"
+				if(2)
+					target.gender = pick("Male", "Female", "Other")
+				if(3)
+					target.age = rand(5, 85)
+				if(4)
+					target.wanted_status = pick(WANTED_STATUSES)
+				if(5)
+					target.species = pick(get_selectable_species())
+			continue
+
+		else if(prob(1))
+			qdel(target)
+			continue
```

Comment:
this doesn't make sense.
This means EMP'ing your computer having Discord app will fuck up the whole Discord user DB.